server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -588,6 +596,44 @@ private static void parseNonDynamicArray(DocumentParserContext context, final St
                 parseValue(context, lastFieldName);
             }
         }
+        postProcessDynamicMappers(context);
+    }
+
+    private static void postProcessDynamicMappers(DocumentParserContext context) {
+        if (context.indexSettings().getIndexVersionCreated().onOrAfter(DYNAMICALLY_MAP_DENSE_VECTORS_INDEX_VERSION)) {
+            List<Mapper> postProcessedMappers = new ArrayList<>();
+
+            Set<String> dynamicDenseVectorFieldNames = context.getDynamicMappers()
+                .stream()
+                .filter(subClassObj -> subClassObj instanceof NumberFieldMapper)
+                .map(NumberFieldMapper.class::cast)
+                .filter(m -> "float".equals(m.typeName()) && context.isFieldAppliedFromTemplate(m.name()) == false)
+                .collect(Collectors.groupingBy(FieldMapper::name, Collectors.counting()))
+                .entrySet()
+                .stream()
+                .filter(e -> e.getValue() >= MIN_DIMS_FOR_DYNAMIC_FLOAT_MAPPING && e.getValue() <= MAX_DIMS_COUNT)
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toSet());
+
+            if (dynamicDenseVectorFieldNames.isEmpty()) {
+                return;
+            }
+
+            for (Mapper mapper : context.getDynamicMappers()) {
+                if (dynamicDenseVectorFieldNames.contains(mapper.name())) {
+                    DenseVectorFieldMapper.Builder builder = new DenseVectorFieldMapper.Builder(
```

Comment:
As it stands, even on a "global post process" outside of the DocumentParser, we create a new `dense_vector` mapper for every float, so we create, potentially 2048 more objects than is strictly needed. If we know all these things are floats and they are all for the same field name, we should be able to take a count and do a single dynamic mapper update for the dense vector builder